kclvm/driver/src/kpm_update.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::{bail, Result};
+use std::{collections::HashMap, env, iter, path::PathBuf, process::Command};
+
+const MANIFEST_FILE: &str = "kcl.mod";
+
+/// Update the KCL module.
+///
+/// This function calls `kcl mod update` to update the KCL module.
+pub(crate) fn update_kcl_module(manifest_path: PathBuf) -> Result<()> {
+    match lookup_the_nearest_file_dir(manifest_path.clone(), MANIFEST_FILE) {
+        Some(mod_dir) => {
+            match Command::new(kcl())
+                .arg("mod")
+                .arg("update")
+                .current_dir(mod_dir)
+                .output()
+            {
+                Ok(output) => {
+                    if !output.status.success() {
+                        bail!(
+                            "update failed with error: {}",
+                            String::from_utf8_lossy(&output.stderr)
+                        );
+                    }
+                    Ok(())
+                }
+                Err(err) => bail!("update failed with error: {}", err),
+            }
+        }
+        None => bail!("Manifest file '{}' not found in directory hierarchy", MANIFEST_FILE),
+    }
+}
+
+/// Get the path for the KCL executable.
+fn kcl() -> PathBuf {
+    get_path_for_executable("kcl")
+}
+
+/// [`get_path_for_executable`] will return the path for [`executable_name`].
+pub fn get_path_for_executable(executable_name: &'static str) -> PathBuf {
+    // The current implementation checks $PATH for an executable to use:
+    // `<executable_name>`
+    //  example: for <executable_name>, this tries just <executable_name>, which will succeed if <executable_name> is on the $PATH
+
+    if lookup_in_path(executable_name) {
+        return executable_name.into();
+    }
+
+    executable_name.into()
+}
+
+/// [`lookup_in_path`] will search for an executable file [`exec`] in the environment variable ‘PATH’.
+///  If found, return true, otherwise return false.
+fn lookup_in_path(exec: &str) -> bool {
+    let paths = env::var_os("PATH").unwrap_or_default();
+    env::split_paths(&paths)
+        .map(|path| path.join(exec))
+        .find_map(probe)
+        .is_some()
+}
+
+/// [`probe`] check if the given path points to a file.
+/// If it does, return [`Some`] of the path.
+/// If not, check if adding the current operating system's executable file extension (if any) to the path points to a file.
+/// If it does, return [`Some`] of the path with the extension added.
+/// If neither, return [`None`].
+fn probe(path: PathBuf) -> Option<PathBuf> {
```

Comment:
The same as the method kcl(), you can move it to https://github.com/kcl-lang/kcl/blob/main/kclvm/driver/src/lib.rs